### ISSUE TITLE
[REQUEST] `on_component` for v4

### ISSUE THREAD

**Issue by Toricane** [interactions-py/interactions.py#481]
**What is the feature you're proposing?**
I would really like it if we could introduce `on_component` back into this library.
They used to be in v3, which was an event that was triggered on any component interactions.
Useful if for example, custom_ids of components had extra information in them, so component callbacks can't be used.
There is already this:
```py
@bot.event
async def on_interaction_create(event: interactions.Interaction):
  if event.type == interactions.InteractionType.MESSAGE_COMPONENT:
    ...
```
However, it would make more sense if there was something like this:
```py
@bot.event
async def on_component(ctx: interactions.ComponentContext):
    ...
```

**Additional Information**
- [ ] My feature request is related to an existing Issue.
    - Issue (if referenceable):

**Comment by Toricane** [comment]
Credit to The Italian Government in #lib-discussion

**Comment by i0bs** [comment]
I'll be looking into this request after #451 has been successfully merged.

**Comment by r-warfield** [comment]
now that we have modals an on_modal would be good as well
should i make another issue or just keep this here

**Comment by i0bs** [comment]
`on_modal`, `on_autocomplete`, `on_component`, `on_command` and `on_interaction` have all been added as of #452 .
